### ISSUE TITLE
No RPC exception is thrown when a received Response object does not contain both the "result" and "error" fields

### ISSUE THREAD

**Issue by debevv** [elpheria/rpc-websockets#91]
Hi, I was using the client in the library against a simple server that echoes the JSON message sent by the client, for example:
``` json
{
  "jsonrpc":"2.0",
  "method":"runScript",
  "params":["test"],
  "id":4
}
```
Per JSON-RPC 2.0 spec, the server is required to respond with a `Response` object which contains a `result` property in case of success, or an `error` on in case of error. But, when my server responses with that JSON, both of them are absent.
So I expected to receive an RPC exception when making a call() and getting this as response. Instead, the method simply resolves with an `undefined` result.
Is this intended behavior?

**Comment by mkozjak** [comment]
Hi, @debevv, I’m not 100% sure if I understood what sxactly your server sends to the client, so could you provide the code example using this lib’s server class so I can take a closer look? Thanks!

**Comment by debevv** [comment]
Sorry, I should have phrased it better.
Essentially, using the library as a client, if i call() a method:
```
var result = await ws.call(method, params)
```
and the server responds with an RPC Response object like this (which should be non-compliant):
```
{
  "jsonrpc":"2.0",
  "id":4
}
```
the `result` variable will be `undefined`, and no exception will be thrown by `call()`. I would expect an exception here because of this JSON-RPC 2.0 spec statement: "Either the result member or error member MUST be included, but both members MUST NOT be included."

**Comment by mkozjak** [comment]
The response you provided is a valid one, it's compliant. The json-rpc 2.0 rule you quoted is handled internally by rpc-websockets here https://github.com/elpheria/rpc-websockets/blob/master/src/lib/client.ts#L300.
Again, I'd need a complete code example for both server (whatever language) and client (here) to see if we're having a bug, eventually.

**Comment by mkozjak** [comment]
@debevv - Aah, thanks for the explanation. Makes sense. Are you able to provide a full example of your server code (any language) and client code (here)?

**Comment by debevv** [comment]
This is the server code, written in go: [ws-server.zip](https://github.com/elpheria/rpc-websockets/files/5220099/ws-server.zip)
The client code is very simple indeed, just connect and make a call(), that is basically what I'm doing at the moment in my web client:
``` typescript
const WSClient = require("rpc-websockets").Client

class RPCSocket {
    private static socket: typeof WSClient

    public static connect() {
        if (this.socket) {
            this.socket.close()
        }

        this.socket = new WSClient("ws://localhost:2601", {
            autoconnect: false,
            reconnect: true,
            max_reconnects: 0
        })

        return new Promise((resolve, reject) => {
            this.socket.on("error", (error: Error) => { reject(error) })
            this.socket.on("open", () => {
                this.socket.removeListener("error")
                resolve()
            })

            this.socket.connect()
        })
    }

    static async call(method: string, params: string[]) {
        return this.socket.call(method, params, 10000)
    }
}

RPCSocket.connect()
const result = RPCSocket.call("runScript", [ "test" ])
console.log(result)
```

**Comment by mkozjak** [comment]
Thanks, @debevv, this should be fixed in 7.4.3.
